textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/keywordmatch/KeywordMatcher.java

```diff
@@ -124,10 +140,62 @@ public ITuple getNextTuple() throws DataFlowException {
                     }
 
                     boolean allTokenPresent = areAllTrue(tokensPresent);
+                    if(predicate.getOperatorType() == KeywordOperatorType.BASIC) {
```

Comment:
spanList contains all the token spans in the documents in all textFields as returned by lucene index.
span.getFieldName().equals(fieldName) is used to check if the span being checked is in same attribute. if it is we make sure all the tokens are present in one attribute, otherwise we need to remove them from the spanlist.

Does this helps, or should I try to elaborate my point with an example.